### ISSUE TITLE
Last item on tabs is covered by arrow

### ISSUE THREAD

**Issue by anthonydillon** [canonical-web-and-design/vanilla-framework#3602]
![image](https://user-images.githubusercontent.com/1413534/109662783-57cd8e80-7b63-11eb-9bec-e49b000b33e4.png)

https://vanillaframework.io/docs/examples/patterns/tabs

**Comment by clagom** [comment]
On mobile it looks like this
![IMG_0948.PNG](https://images.zenhubusercontent.com/5b0d889d89d5dc41604fa14c/bdcca812-aaa3-49af-9a34-86e8db3383c9)

**Comment by cristinadresch** [comment]
Some discussions:

- [Lyubo] Are there any accessibility requirements? If not, then we should be able to get rid of the arrow
- [Will] We could always show the blur so it looks like there's more nav to the right
- [Wayne] If we keep the arrow so it slides to the right, so we need to add an arrow to the left too?
- [Elio] Also if there's an arrow and you go all the way to the right the arrow should disappear

To do:

- File a bug for the arrow issue from Claudio's screenshot @clagom 
- UX person to be assigned to this, to perform benchmarking and accessibility work @elioqoshi @long-chung

**Comment by cristinadresch** [comment]
@elioqoshi @long-chung as discussed in the last vanilla sync, I'm assigning this issue to this iteration. If you two could discuss how to split the work etc. keeping in mind Long is off next week. cc @SirSamTumless

**Comment by elioqoshi** [comment]
@cristinadresch Thanks. I think I should be able to do the majority here with @long-chung maybe doing a review afterwards if that sounds okay.

**Comment by elioqoshi** [comment]
Trying to spec this out here firstly as it was not trivial to create this interaction in a prototyping tool.

I did some benchmarking and the IBM Carbon Design System is close to what we discussed this component could be like:
https://www.carbondesignsystem.com/components/tabs/usage/#live-demo

## Desktop Preview

https://user-images.githubusercontent.com/5436686/112859846-85025380-90b3-11eb-9b2e-0610a2a03b09.mov
- Active and hover item are visually distinct. I do think we should do the same to ensure these are distinguishable. As of now if you hover over an inactive item, it can give the impression that it's the active item.

## Mobile Preview (Drag)

https://user-images.githubusercontent.com/5436686/112860040-bc710000-90b3-11eb-88ec-0c657bab52b6.mov

- Dragging/horizontal scrolling works fine, though there are some snapping errors once you reach the end or go back to the beginning again where the items shift.
- Arrows indicate whether there is more content available.

## Mobile Preview (Clicking on Arrow)

https://user-images.githubusercontent.com/5436686/112860443-1c67a680-90b4-11eb-80da-8378b2788aea.mov

- Clicking on arrows moves the view a little bit in that direction, which feels awkward. The view should be shifted exactly to the next item instead.

## Other Considerations

- We might want to highlight the active item more. Using a larger font-weight for active and Light weight for other tab items could help with that.
- In order to address the original issue here, the tab group should have more padding left and right in order to allow some space for the arrows without overlapping the items.

**Comment by long-chung** [comment]
Is the idea that all tabs will be the same width?

**Comment by elioqoshi** [comment]
> Is the idea that all tabs will be the same width?

That's just an idea. As I said it would ensure we have equal emphasis (mostly by touch area) for every item. It could just be a different flavor though and we can keep both. It's something I raised so we can discuss it.

**Comment by elioqoshi** [comment]
Here is how that would look like for a tab group with 4 items:

![grafik](https://user-images.githubusercontent.com/5436686/112953569-e3741400-913d-11eb-802f-51cef5112fcf.png)


I'd argue that using a brand colour or another semantic colour would help even more:

![grafik](https://user-images.githubusercontent.com/5436686/112953702-fd155b80-913d-11eb-92b4-dda6b8260aaf.png)

**Comment by elioqoshi** [comment]
Here is how that would look like with the (more or less) the current layout:

![grafik](https://user-images.githubusercontent.com/5436686/112956354-a9584180-9140-11eb-82e7-a5dbc4e65282.png)

**Comment by clagom** [comment]
We discussed separating @elioqoshi's exploration in two parts (_alignment of labels and tabs_ and _use of arrow_). Elio's benchmarking seems to validate the use of the arrow. We are proposing to remove the scrollbar and making the arrow to act on click/tap/scroll, also fixing the current visual bug on mobile.

**Comment by clagom** [comment]
@long-chung to provide more examples and benchmarking to validate the case of not using the arrow.

**Comment by long-chung** [comment]
Miro board with some examples of high frequency of visitors/users mobile sub navigation.
https://miro.com/app/board/o9J_lMEgL20=/

**Comment by clagom** [comment]
Thanks @long-chung I think we need to work on the "whys" too. Pro and cons of having an arrow or not? What is the UX, use cases that push us to go with a solution rather than the other? CC @elioqoshi @lyubomir-popov 
Personally, I don't really have a strong opinion but I don't see a strong reasoning *not to have* arrows to reinforce the scroll call to action (in your examples, Dribbble). I'd remove the current scroll bar anyway, because it's redundant with the arrows.

**Comment by long-chung** [comment]
Thanks Claudio.
My rationale for why we shouldn't have arrows in the navigation are as follows:

- Mobile screens are already quite narrow, so taking up an additional X amount of pixels would be better served to show more of the label/s or fit more on screen
- The natural tendancy on mobile is a swipe or scroll action, from scrolling pages to actioning events ala email clients, native app behaviours. This you could argue has been ingrained into users behavioural modals on a mobile experience.
https://uxplanet.org/horizontal-scrolling-in-mobile-643c81901af3

**Comment by elioqoshi** [comment]
> The natural tendancy on mobile is a swipe or scroll action, from scrolling pages to actioning events ala email clients, native app behaviours. This you could argue has been ingrained into users behavioural modals on a mobile experience.
https://uxplanet.org/horizontal-scrolling-in-mobile-643c81901af3

I am not sure if I understood your point correctly. The article lists various perspectives on the topic from what I can see.

Labels being cut by the viewport or fading out by a light gradient is not sufficient to indicate there is more content. In its default state, the arrow would only be present on the right side, since the user just entered the tab view. Not having that can lead to users not understanding there is more content beyond the viewport (especially if the labels have such a length they are not covered by the gradient or aren't cut off by the viewport), so that means that users will see left and right arrows only in the case once they interacted with the component. In my understanding, that flows into an accessibility enhancement over having no arrows. Also don't forget that this pattern won't be exclusive on mobile, but will exist also on Desktop.

There are a few things we could do here at this point:
1. Spend some time with A/B testing different scenarios and understand which pattern is performing better in which circumstances (too much effort in my opinion for a component with limited impact as of now).
2. Implement both flavours in Vanilla and develop guidelines when to use what (this would make A/B testing easier though).
3. We go with either of these options and revisit later to identify any pitfalls.

What do you think?

**Comment by elioqoshi** [comment]
Apple, Google and Amazon have subpar accessibility practices in their design systems however, because they can afford to do that. Vanilla strives to be fully accessible whereas Material Design and co ignore certain accessibility guidelines on purpose.

@cassiocassio would this research be something the research team would be able to conduct in the near future or are there more relevant things to tackle first?

**Comment by clagom** [comment]
Can we return to the original point of this issue @long-chung @elioqoshi 
The problem here is that the last item still displays the arrow, as if there's more to the right, and the arrow also cover the last item of the tabs.
During standup @anthonydillon pointed out that this component is CSS-only and doesn't use any javascript, hence the odd behaviour.
We have seen from @long-chung's quick benchmarking that not using arrows is acceptable, and that users are informed about more content to scroll via trunctation of the text/items. I would then suggest to go this way, removing arrows and scrollbars, keeping the component JS free and UX validated from the benchmarking.

@long-chung, was this solution implemented on the JAAS Dashboard already?

If we want to iterate this componen afterwards to use the arrow, this would require JS as the arrows should disappear while displaying first and last items without covering tab items.

CC @bartaz @SirSamTumless @lyubomir-popov 

What do we think? Can we unblock?

**Comment by long-chung** [comment]
@anthonydillon has removed the arrow on the JAAS dashboard and what we plan on using moving forward.
Happy to revisit this at a later date if/when we come up with a different solution.

**Comment by clagom** [comment]
Thanks @long-chung. Can we then adopt those changes done in the JAAS Dashboard in Vanilla too, as a first step to close this issue? @bartaz @lyubomir-popov @sowasred2012

**Comment by lyubomir-popov** [comment]
No objecions on my end. @bartaz @sowasred2012 if you agree then this issue can go straight to build?

**Comment by sowasred2012** [comment]
@lyubomir-popov @clagom yeah - I think it makes sense.

**Comment by bartaz** [comment]
>  I would then suggest to go this way, removing arrows and scrollbars, keeping the component JS free and UX validated from the benchmarking.

We can easily remove the arrow, but we can't just drop the scrollbars. To make all the tabs accessible on narrow screens they need to be scrollable and for that they need scrollbar.
Scrollbar will only appear when needed (when content is wider than the screen) and (usually, depending on the browser) only during the scroll.

**Comment by clagom** [comment]
Sounds good that's reasonable with only CSS. Thank you for pointing that out, @bartaz
